force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -102,16 +113,14 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer countVowels(String str) {
         // Initialize the result Integer
-
-        /*
-        for () {
-            // get individual characters from the string - hint use substring or split
-            // check if the character is a vowel
-            // if it is a vowel, increment the result Integer
-        }
-        */
-
-        return null; // Replace null with the variable you used to store the result
+        Integer finalVowelCount = 0;
+        for (Integer i = 0; i < str.length(); i++) {
+            String lwrSentence = str.substring(i, i+1).toLowerCase();
+            if (lwrSentence.contains('a')|| lwrSentence.contains('e')|| lwrSentence.contains('i')|| lwrSentence.contains('o')|| lwrSentence.contains('u')) {
+                finalVowelCount++;
+            }
+        } 
+        return finalVowelCount; // Replace null with the variable you used to store the result
```

Comment:
For problems like this, where you are checking for membership, I always recommend using a set. Because set values are hashed, it is the most permanent way to check for membership and is easier to read than conditional logic. 
```
  public static Integer countVowels(String str) {
        Integer vowelCount = 0;
        Set<String> vowels = new Set<String>{
                'a', 'e', 'i', 'o', 'u'
        };
        List<String> characters = str.split('');
        for (String character : characters) {
            if (vowels.contains(character.toLowerCase())) {
                vowelCount++;
            }
        }
        return vowelCount;
    }
    ```